lib/promscrape/targetstatus.go

```diff
@@ -355,6 +360,25 @@ func (tsm *targetStatusMap) getTargetsStatusByJob() ([]jobTargetsStatuses, []str
 	return jts, emptyJobs
 }
 
+func filterJobs(jts []jobTargetsStatuses, searchQuery string) []jobTargetsStatuses {
+	regQuery := filterReg.ReplaceAllString(searchQuery, ")(?=.*")
+	finder := regexp.MustCompile(regQuery)
+
+	var filteredJts []jobTargetsStatuses
+	var ts []targetStatus
+	for _, job := range jts {
+		for _, targetStatus := range job.targetsStatus {
+			if finder.FindString(targetStatus.sw.Config.ScrapeURL) != "" ||
+				finder.FindString(targetStatus.sw.Config.LabelsString()) != "" {
+				ts = append(ts, targetStatus)
```

Comment:
looks like `ts` keeps the state across multiple jobs, constantly growing. This would produce incorrect results on /targets page.

Try to apply filter `instance="demo.promlabs.com:10000"` to the following config:
```
  - job_name: 'promlabs'
    static_configs:
      - targets:
          - 'demo.promlabs.com:10000'
          - 'demo.promlabs.com:10001'
          - 'demo.promlabs.com:10002'
```

Reply:
updated logic. found bug which you described)